tests/src/test/java/org/jboss/test/marshalling/SimpleMarshallerTests.java

```diff
@@ -3660,4 +3665,72 @@ public void runRead(Unmarshaller unmarshaller) throws Throwable {
             }
         });
     }
+
+    @Test(description = "JBMAR-222_vector")
+    private void testConcurrentVector() throws Throwable {
+        testConcurrentAccess(new Vector(), Vector.class);
+    }
+
+    @Test(description = "JBMAR-222_stack")
+    private void testConcurrentStack() throws Throwable {
+        testConcurrentAccess(new Stack(), Stack.class);
+    }
+
+    private void testConcurrentAccess(Collection instance, Class<? extends Collection> collectionClass) throws Throwable {
+        //if versin is not -1, we have to assert whether run wasn't fail
+        if((testMarshallerProvider instanceof  MarshallerFactoryTestMarshallerProvider && ((MarshallerFactoryTestMarshallerProvider)testMarshallerProvider).getVersion() == -1) ||
+                (testUnmarshallerProvider instanceof  MarshallerFactoryTestUnmarshallerProvider && ((MarshallerFactoryTestUnmarshallerProvider)testUnmarshallerProvider).getVersion() == -1)) {
+            throw new SkipException("Test not relevant for configuration version -1");
+        }
+
+        instance.add(new SlowSerialize());
+        AtomicBoolean failed = new AtomicBoolean(false);
+        StringBuilder sb = new StringBuilder();
+        // Kick off serialization
+        Thread t = new Thread() {
+            public void run () {
+                try {
+                    runReadWriteTest(new ReadWriteTest() {
+                        @Override
+                        public void runWrite(Marshaller marshaller) throws Throwable {
+                            marshaller.writeObject(instance);
+                        }
+
+                        @Override
+                        public void runRead(Unmarshaller unmarshaller) throws Throwable {
+                            Collection result = unmarshaller.readObject(collectionClass);
+                            assertEquals("JBMAR-222: ", 1,  result.size());
+                        }
+                    });
+                } catch ( Throwable t ) {
+                    failed.set(true);
+                    t.printStackTrace();
+                    StringWriter writer = new StringWriter();
+                    PrintWriter printWriter= new PrintWriter(writer);
+                    t.printStackTrace(printWriter);
+                    sb.append(t.toString()).append(writer.toString());
+                }
+            }
+        };
+        t.start ();
+
+        // Wait for serialization to start, then modify the Vector
+        try { Thread.sleep ( 500 ); } catch ( InterruptedException e ) {        }
```

Comment:
Sleeps are not the right approach here; instead a synchronization primitive of some sort must be used.  However in this case I think it's not worth the work: it can be statically verified that the synchronization occurs, and it's not really possible to prove (in any way) that there aren't deadlocks and so forth.  So I think the test can be skipped.

Reply:
Test eliminated @dmlloyd , thanks for the input.